doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -133,47 +116,65 @@ Programmable proof search
          Goal forall P:nat -> Prop, P 0 -> exists n, P n.
          eauto.
 
-      Note that ``ex_intro`` should be declared as a hint.
+      `ex_intro` is declared as a hint so the proof succeeds.
 
+   .. seealso:: :ref:`thehintsdatabasesforautoandeauto`
 
-   .. tacv:: {? info_}eauto {? @natural} {? using {+ @qualid}} {? with {+ @ident}}
+   .. tacn:: info_eauto {? @int_or_var } {? @auto_using } {? @hintbases }
 
-      The various options for :tacn:`eauto` are the same as for :tacn:`auto`.
+      The various options for :tacn:`info_eauto` are the same as for :tacn:`info_auto`.
 
    :tacn:`eauto` also obeys the following flags:
 
    .. flag:: Info Eauto
              Debug Eauto
       :undocumented:
 
-   .. seealso:: :ref:`The Hints Databases for auto and eauto <thehintsdatabasesforautoandeauto>`
+   .. tacn:: debug eauto {? @int_or_var } {? @auto_using } {? @hintbases }
 
+      Behaves like :tacn:`eauto` but shows the tactics it tries to solve the goal,
+      including failing paths.
+
+   .. tacn:: bfs eauto {? @int_or_var } {? @auto_using } {? @hintbases }
+
+      Like :tacn:`eauto`, but uses a
+      `breadth-first search <https://en.wikipedia.org/wiki/Breadth-first_search>`_.
 
-.. tacn:: autounfold with {+ @ident}
-   :name: autounfold
+.. tacn:: autounfold {? @hintbases } {? @occurrences }
 
-   This tactic unfolds constants that were declared through a :cmd:`Hint Unfold`
+   Unfolds constants that were declared through a :cmd:`Hint Unfold`
    in the given databases.
 
-.. tacv:: autounfold with {+ @ident} in @goal_occurrences
+   :n:`@occurrences`
+     Performs the unfolding in the specified occurrences.  The :n:`at @occs_nums`
+     clause of :n:`@occurrences` is not supported.
+
+.. tacn:: autorewrite {? * } with {+ @ident } {? @occurrences } {? using @ltac_expr }
+
+   `*`
+     If present, rewrite all occurrences.
```

Comment:
Add the following so that it approximately reflects the intent:
```suggestion
     If present, rewrite all occurrences whose side-conditions are solved.
```

And maybe add a `.. todo` as well to signal that this may not work as described, with a reference to the issues Jason linked to.